apps/mdm/tasks.py

```diff
@@ -55,8 +55,7 @@ def update_existing_devices(fleet: Fleet, mdm_devices: list[dict]):
     }
 
     our_devices = Device.objects.filter(
-        Q(fleet=fleet)
-        & (Q(device_id__in=devices_by_id.keys()) | Q(serial_number__in=devices_by_serial.keys()))
+        Q(device_id__in=devices_by_id.keys()) | Q(serial_number__in=devices_by_serial.keys())
```

Comment:
I see, so by excluding the Fleet from our filter, we get all matching devices. Then the device in the wrong fleet will get updated to the correct fleet, rather than creating a new device?

Reply:
Good point, sounds good!